client/src/components/teacher/TeacherView.jsx

```diff
@@ -56,13 +63,47 @@ function TeacherView() {
     setSelectedCourse(course);
   };
 
-  const handleOpen = () => setIsAddingCourse(true);
-  const handleClose = () => setIsAddingCourse(false);
+  const handleCreateGroup = async () => {
+    const response = await request("/courses/create-group", "POST", {
+      course_id: selectedCourse._id,
+    });
+    return;
+  }
+
+  // handleOpen performs different functions depending on string parameter
+  const handleOpen = (mode) => {
+    switch (mode) {
+      case 'course': {
+        setIsAddingCourse(true);
+        break;
+      }
+      case 'group': {
+        setIsCreatingGroup(true);
+        break;
+      }
+      default: break;
+    }
+    return;
+  }
+  const handleClose = (mode) => {
+    switch(mode) {
+      case 'course': { 
+        setIsAddingCourse(false);
+        break;
+      } 
+      case 'group': {
+        setIsCreatingGroup(false);
+        break;
+      }
+      default: break;
+    }
+    return;
+  }
```

Comment:
good handling of different modals open/close function while keeping a single function